src/lib.rs

```diff
@@ -19,12 +19,23 @@ pub mod worktree;
 /// The bool in the return value specifies whether there is a repository
 /// in root itself.
 #[allow(clippy::type_complexity)]
-fn find_repos(root: &Path) -> Result<Option<(Vec<repo::Repo>, Vec<String>, bool)>, String> {
+fn find_repos(
+    root: &Path,
+    exclusion_pattern: &Option<String>,
+) -> Result<Option<(Vec<repo::Repo>, Vec<String>, bool)>, String> {
     let mut repos: Vec<repo::Repo> = Vec::new();
     let mut repo_in_root = false;
     let mut warnings = Vec::new();
 
+    let exlusion_regex: regex::Regex =
+        regex::Regex::new(&exclusion_pattern.clone().unwrap_or(r"^$".to_string()))
+            .unwrap_or_else(|_| regex::Regex::new(r"^$").unwrap());
```

Comment:
In case of an invalid regex, you'd silently discard the error and fall back to `^$`. I think it would be preferable to abort early with a proper error message, so the user can fix their regex.

You could do it like this:

`regex::Regex::new()` returns `Result<Regex, Error>`. If we get an error, we could immediately return this by using [the `?` operator](https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html#a-shortcut-for-propagating-errors-the--operator). But `find_repos()` returns `Result<..., String>`. How can we convert this?

Take a look at [`Result::map_err()`](https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err). It applies a function to the `Err` value of a `Result`, and leaves the `Ok` variant unmodified. Perfect for us.

So, you could change the code as follows:

```suggestion
            .map_err(|e| format!("invalid regex: {e}"))?;
*